packages/samples/headless-ssr-commerce/hooks/use-recent-queries.ts

```diff
@@ -0,0 +1,49 @@
+import {recentQueriesMaxLength} from '@/shared/recent-queries-config';
+import {useEffect} from 'react';
+
+const recentQueriesKeyStorageKey = 'commerce-recent-queries';
+
+function getStoredRecentQueries(): string[] {
+  const storedQueries = localStorage.getItem(recentQueriesKeyStorageKey);
+  if (storedQueries) {
+    try {
+      return JSON.parse(storedQueries);
+    } catch (error) {
+      console.error('Failed to parse recent queries from localStorage:', error);
+    }
+  }
+  return [];
+}
+
+function saveRecentQueries(queries: string[]) {
+  localStorage.setItem(
+    recentQueriesKeyStorageKey,
+    JSON.stringify(queries.slice(0, recentQueriesMaxLength))
```

Comment:
I don't think we need to slice the queries here: they will be sliced (and duplicates will be removed) when the registerQueries action is handled in the recent queries slice.